tree/dataframe/src/RLoopManager.cxx

```diff
@@ -766,24 +788,81 @@ void RLoopManager::Jit()
 {
    {
       R__READ_LOCKGUARD(ROOT::gCoreMutex);
-      if (GetCodeToJit().empty()) {
+      if (GetCodeToJit().empty() && GetCodeToDeclare().empty()) {
+         RunDeferredCalls();
          R__LOG_INFO(RDFLogChannel()) << "Nothing to jit and execute.";
          return;
       }
    }
 
-   const std::string code = []() {
+   std::string codeToDeclare, code;
+   {
       R__WRITE_LOCKGUARD(ROOT::gCoreMutex);
-      return std::move(GetCodeToJit());
-   }();
+      codeToDeclare.swap(GetCodeToDeclare());
+      code.swap(GetCodeToJit());
+   };
 
    TStopwatch s;
    s.Start();
-   RDFInternal::InterpreterCalc(code, "RLoopManager::Run");
+   if (!codeToDeclare.empty()) {
+      // This takes care of autoloading when needed
+      // If instead we used gInterpreter->Declare, builds with runtime_cxxmodules=OFF would fail in
+      // jitted actions with custom helpers with errors like:
+      // error: 'MyHelperType' is an incomplete type
+      // return std::make_unique<Action_t>(Helper_t(std::move(*h)), bl, std::move(prevNode), colRegister);
```

Comment:
Does that result into a different performance between modules and non-modules? If so you can check if you have modules on using `HasPCMForLibrary` and do different things in both cases...